File: algorithm/53.最大子序和.cpp
#include "iostream"
#include "string"
#include "vector"
using namespace std;

/**
 * 
 */
class Solution {
public:
  int maxSubArray(vector<int>& nums) {
    if (nums.empty()) return 0;
    vector<int> res(nums.size());
    res[0] = nums[0];
    for (unsigned i = 1; i < nums.size(); i++) {
      res[i] = max(nums[i], res[i - 1] + nums[i]);
    }
  }
};
int main(void) {

  return 0;
}

File: algorithm/53.最大子序和.js
/**
 * @param {number[]} nums
 * @return {number}
 */
// 这题是求最大连续子序和 如果用下面这种递归的话 那求的是 最大和
// var maxSubArray = function(nums) {
//   const _maxSubArray = (start) => {
//     if (start >= nums.length) return 0;
//     const a = nums[start] + _maxSubArray(start + 1);
//     const b = _maxSubArray(start + 1);
//     return Math.max(a, b);
//   }
//   return _maxSubArray(0);
// };

// 动态规划
var maxSubArray = function(nums) {

};
